### ISSUE TITLE
Stacktrace info is empty in safari 

### ISSUE THREAD

**Issue by brianshen1990** [bugsnag/bugsnag-js#1840]
### Describe the bug

We are using bugsnag in FE, while we get stacktrace info in Chrome. But in safari, the stacktrace info is alway empty. 

### Steps to reproduce
1. Use Safari to go to https://martial-art-jan24.account.myshopify.com/info
2. Use any email to login
3. go to https://martial-art-jan24.account.myshopify.com/orders/4804755849238
4. open dev tool, block graphqls sent to `https://atlas.shopifysvc.com/graphql` 
5. See error
<img width="1682" alt="Screen Shot 2022-10-19 at 15 12 16" src="https://user-images.githubusercontent.com/12724940/196782911-6e964220-9c4f-451b-8121-c8bc447aba49.png">

### Environment
* Bugsnag version:
* Browser framework version (if any):
    * Vue:
    * React: *
    * Angular:
* Server framework version (if any):
    * Koa:
    * Express:
    * Restify:
* Browser version (e.g. chrome, safari): Safari
* Device (e.g. iphonex): MacOS

### Example Repo 

- [ ] Create a minimal repository that can reproduce the issue
- [ ] Link to it here:

### Example code snippet 

```
# (Insert code sample to reproduce the problem)
```

<details><summary>Error messages:</summary>

```

```
</details>

**Comment by luke-belton** [comment]
Hi @brianshen1990 - please could you write into support@bugsnag.com, referencing this GitHub issue and sharing links to examples of the events from Chrome/Safari in your Bugsnag dashboard? We'd be happy to help you out from there and will update this thread with any outcomes for future travellers 👍

**Comment by johnkiely1** [comment]
This has been resolved via the support ticket. 

For the benefit of future travellers: 

We did some testing and it looks like for these kinds of unhandled promise rejections, Safari (and also Firefox) does not provide a stacktrace that we can report.
 
For some context, these errors are detected by Bugsnag's unhandled rejection handler: https://github.com/bugsnag/bugsnag-js/blob/next/packages/plugin-window-unhandled-rejection/unhandled-rejection.js 
 
Bugsnag attaches an event listener to the window's unhandledrejection event, and creates an error event from the error object that is passed to this event listener. In Safari and Firefox, the error that is passed is TypeError: failed to load , but that error object does not contain a stacktrace.
 
When events do not contain a stacktrace, Bugsnag does make attempts to generate one:
https://github.com/bugsnag/bugsnag-js/blob/936bc57e30367388d83702bbe7ef18f32474fbcc/packages/core/event.js#L168-L178 
https://github.com/bugsnag/bugsnag-js/blob/936bc57e30367388d83702bbe7ef18f32474fbcc/packages/core/event.js#L281-L294 
 
But unfortunately this sometimes isn't possible, and particularly for unhandled promise rejections (both in the browser and in node), we are very limited in what we can do to get a stacktrace.
 
In cases like this we'd recommend looking at the event's breadcrumbs and metadata to help identify the cause.
